# Analysis of World Happiness Report
#### Studying the various factors that might affect a countries Happiness
Source: https://www.kaggle.com/mathurinache/world-happiness-report

In [1]:
import pandas as pd 

data_2015 = pd.read_csv('2015.csv')
data_2016 = pd.read_csv('2016.csv')
data_2017 = pd.read_csv('2017.csv')
data_2018 = pd.read_csv('2018.csv')
data_2019 = pd.read_csv('2019.csv')
data_2020 = pd.read_csv('2020.csv')

## 2015 World Happiness Report Analysis

In [2]:
import plotly.express as px
import plotly.graph_objects as go


data_2015.columns

Index(['Country', 'Region', 'Happiness Rank', 'Happiness Score',
       'Standard Error', 'Economy (GDP per Capita)', 'Family',
       'Health (Life Expectancy)', 'Freedom', 'Trust (Government Corruption)',
       'Generosity', 'Dystopia Residual'],
      dtype='object')

### Details of Columns
-  <b>Country</b>: Name of the Country
-  <b>Region</b>: Region the Country belongs to
-  <b>Happiness Rank</b>: Rank of the country based on the Happiness Score.
-  <b>Happiness Score</b>: A metric measured in 2015 by asking the sampled people the question: "How would you rate your happiness on a scale of 0 to 10 where 10 is the happiest."
-  <b>Standard Error</b>: The standard error of the happiness score.
-  <b>Economy (GDP per Capita)</b>: The extent to which GDP contributes to the calculation of the Happiness Score.
-  <b>Family</b>: The extent to which Family contributes to the calculation of the Happiness Score
-  <b>Health (Life Expectancy)</b>: The extent to which Life expectancy contributed to the calculation of the Happiness Score
-  <b>Freedom</b>: The extent to which Freedom contributed to the calculation of the Happiness Score.
-  <b>Trust (Government Corruption)</b>: The extent to which Perception of Corruption contributes to Happiness Score.
-  <b>Generosity</b>: The extent to which Generosity contributed to the calculation of the Happiness Score.
-  <b>Dystopia Residual</b>: The extent to which Dystopia Residual contributed to the calculation of the Happiness Score.

In [3]:
print('Top 10 Happiest Countries in 2015 :)')
data_2015.Country.head(10)

Top 10 Happiest Countries in 2015 :)


0    Switzerland
1        Iceland
2        Denmark
3         Norway
4         Canada
5        Finland
6    Netherlands
7         Sweden
8    New Zealand
9      Australia
Name: Country, dtype: object

In [4]:
region  = data_2015.Region.head(10)
print(region)

0               Western Europe
1               Western Europe
2               Western Europe
3               Western Europe
4                North America
5               Western Europe
6               Western Europe
7               Western Europe
8    Australia and New Zealand
9    Australia and New Zealand
Name: Region, dtype: object


In [21]:
fig = px.bar(region.value_counts(),title='Which region are these countries from?')
fig.show()

In [6]:
fig = px.bar(data_2015.Region.value_counts(),title='Which region are all countries from?')
fig.show()

In [7]:
print('Total number of countries in Report: ',len(data_2015))

Total number of countries in Report:  158


### Comparision 
To compare the countries they are separated in 5 groups based on their Ranks<br>
<li>Group 1: 1 - 31
<li>Group 2: 32 - 63
<li>Group 3: 64 - 95
<li>Group 4: 96 - 127
<li>Group 5: 128 - 158

In [8]:
group_1 = data_2015[0:31]
group_2 = data_2015[31:63]
group_3 = data_2015[63:95]
group_4 = data_2015[95:127]
group_5 = data_2015[127:]

In [9]:
index=['group_1','group_2','group_3','group_4','group_5']
summary_2015 = pd.DataFrame(data={'Happiness Score':[group_1['Happiness Score'].mean(),
                                          group_2['Happiness Score'].mean(),
                                          group_3['Happiness Score'].mean(),
                                          group_4['Happiness Score'].mean(),
                                          group_5['Happiness Score'].mean()],
                                 'Economy':[group_1['Economy (GDP per Capita)'].mean(),
                                          group_2['Economy (GDP per Capita)'].mean(),
                                          group_3['Economy (GDP per Capita)'].mean(),
                                          group_4['Economy (GDP per Capita)'].mean(),
                                          group_5['Economy (GDP per Capita)'].mean()],
                                 'Family':[group_1['Family'].mean(),
                                          group_2['Family'].mean(),
                                          group_3['Family'].mean(),
                                          group_4['Family'].mean(),
                                          group_5['Family'].mean()],
                                 'Health':[group_1['Health (Life Expectancy)'].mean(),
                                          group_2['Health (Life Expectancy)'].mean(),
                                          group_3['Health (Life Expectancy)'].mean(),
                                          group_4['Health (Life Expectancy)'].mean(),
                                          group_5['Health (Life Expectancy)'].mean()],
                                 'Freedom':[group_1['Freedom'].mean(),
                                          group_2['Freedom'].mean(),
                                          group_3['Freedom'].mean(),
                                          group_4['Freedom'].mean(),
                                          group_5['Freedom'].mean()],
                                 'Trust':[group_1['Trust (Government Corruption)'].mean(),
                                          group_2['Trust (Government Corruption)'].mean(),
                                          group_3['Trust (Government Corruption)'].mean(),
                                          group_4['Trust (Government Corruption)'].mean(),
                                          group_5['Trust (Government Corruption)'].mean()],
                                 'Generosity':[group_1['Generosity'].mean(),
                                          group_2['Generosity'].mean(),
                                          group_3['Generosity'].mean(),
                                          group_4['Generosity'].mean(),
                                          group_5['Generosity'].mean()]},index=index)
summary_2015

Happiness Score   Economy    Family    Health   Freedom     Trust  \
group_1         7.059935  1.283531  1.241786  0.870506  0.580847  0.271536   
group_2         6.041437  1.049322  1.146392  0.751731  0.469687  0.116059   
group_3         5.294219  0.863479  0.987770  0.658382  0.405358  0.111738   
group_4         4.666375  0.604385  0.874978  0.502717  0.319428  0.094552   
group_5         3.820742  0.430653  0.703142  0.367250  0.370702  0.126706   

         Generosity  
group_1    0.300909  
group_2    0.211010  
group_3    0.220594  
group_4    0.220584  
group_5    0.235305

In [10]:
legend = ['group_1','group_2','group_3','group_4','group_5']
px.bar(summary_2015['Happiness Score'],y='Happiness Score',title='Mean Happiness Score:')

In [11]:
names = ['Economy','Family','Health','Freedom','Trust','Generosity']
fig = go.Figure()

for i in legend:
    fig.add_trace(go.Scatter(x=names,y=[summary_2015[j][i] for j in names],name=i))

fig.show()

### Using the above summary we can gain some insights about factors affeting Happiness of a Country in 2015

-  <b>Happiness Score</b>: This is the score used to rank the countries so it makes sense that it decreases as we go down the gropus
-  <b>Economy</b>: This gives us insight that countries with lower GDP per capita tend to be less happy. 
-  <b>Family</b>: Countries with bettter Family life have a better happiness score.
-  <b>Health</b>: Longer life expectancy leads to happier country.
-  <b>Freedom</b>: Freedom doesn't seem to be a great as group_5 is actually above group_4 in Freedom.
-  <b>Trust</b>: Trust or Government Corruption is not great for distinguishing between group_2 to group_4 as they have similar score where as group_1 can be distinguished.
-  <b>Generosity</b>: Generosity also doesn't seem effective for distinguishing between group_2 to group_4 but group_1 is significantly higher 

## 2016 World Happiness Report Analysis

In [12]:
print(data_2015.columns)
print(data_2016.columns)

Index(['Country', 'Region', 'Happiness Rank', 'Happiness Score',
       'Standard Error', 'Economy (GDP per Capita)', 'Family',
       'Health (Life Expectancy)', 'Freedom', 'Trust (Government Corruption)',
       'Generosity', 'Dystopia Residual'],
      dtype='object')
Index(['Country', 'Region', 'Happiness Rank', 'Happiness Score',
       'Lower Confidence Interval', 'Upper Confidence Interval',
       'Economy (GDP per Capita)', 'Family', 'Health (Life Expectancy)',
       'Freedom', 'Trust (Government Corruption)', 'Generosity',
       'Dystopia Residual'],
      dtype='object')


In [13]:
print('Top 10 Happiest Countries in 2016')
data_2016.Country.head(10)

Top 10 Happiest Countries in 2016


0        Denmark
1    Switzerland
2        Iceland
3         Norway
4        Finland
5         Canada
6    Netherlands
7    New Zealand
8      Australia
9         Sweden
Name: Country, dtype: object

In [22]:
region = data_2016.Region.head(10)
fig = px.bar(region.value_counts(),title='Which region are these countries from?')
fig.show()

In [23]:
fig = px.bar(data_2016.Region.value_counts(),title='Which region are all countries from?')
fig.show()

In [16]:
print('Total Number of Countries: ',len(data_2016))

Total Number of Countries:  157


### Comparision 
To compare the countries they are separated in 5 groups based on their Ranks<br>
<li>Group 1: 1 - 31
<li>Group 2: 32 - 63
<li>Group 3: 64 - 95
<li>Group 4: 96 - 127
<li>Group 5: 128 - 157

In [17]:
group_1 = data_2016[0:31]
group_2 = data_2016[31:63]
group_3 = data_2016[63:95]
group_4 = data_2016[95:127]
group_5 = data_2016[127:]

In [18]:
index=['group_1','group_2','group_3','group_4','group_5']
summary_2016 = pd.DataFrame(data={'Happiness Score':[group_1['Happiness Score'].mean(),
                                          group_2['Happiness Score'].mean(),
                                          group_3['Happiness Score'].mean(),
                                          group_4['Happiness Score'].mean(),
                                          group_5['Happiness Score'].mean()],
                                 'Economy':[group_1['Economy (GDP per Capita)'].mean(),
                                          group_2['Economy (GDP per Capita)'].mean(),
                                          group_3['Economy (GDP per Capita)'].mean(),
                                          group_4['Economy (GDP per Capita)'].mean(),
                                          group_5['Economy (GDP per Capita)'].mean()],
                                 'Family':[group_1['Family'].mean(),
                                          group_2['Family'].mean(),
                                          group_3['Family'].mean(),
                                          group_4['Family'].mean(),
                                          group_5['Family'].mean()],
                                 'Health':[group_1['Health (Life Expectancy)'].mean(),
                                          group_2['Health (Life Expectancy)'].mean(),
                                          group_3['Health (Life Expectancy)'].mean(),
                                          group_4['Health (Life Expectancy)'].mean(),
                                          group_5['Health (Life Expectancy)'].mean()],
                                 'Freedom':[group_1['Freedom'].mean(),
                                          group_2['Freedom'].mean(),
                                          group_3['Freedom'].mean(),
                                          group_4['Freedom'].mean(),
                                          group_5['Freedom'].mean()],
                                 'Trust':[group_1['Trust (Government Corruption)'].mean(),
                                          group_2['Trust (Government Corruption)'].mean(),
                                          group_3['Trust (Government Corruption)'].mean(),
                                          group_4['Trust (Government Corruption)'].mean(),
                                          group_5['Trust (Government Corruption)'].mean()],
                                 'Generosity':[group_1['Generosity'].mean(),
                                          group_2['Generosity'].mean(),
                                          group_3['Generosity'].mean(),
                                          group_4['Generosity'].mean(),
                                          group_5['Generosity'].mean()]},index=index)
summary_2016

Happiness Score   Economy    Family    Health   Freedom     Trust  \
group_1         7.012194  1.371737  1.049226  0.789513  0.512259  0.244964   
group_2         6.074406  1.188163  0.939861  0.662696  0.397435  0.125840   
group_3         5.353406  1.012827  0.799904  0.611992  0.344167  0.098377   
group_4         4.627813  0.717034  0.635015  0.441983  0.336058  0.105230   
group_5         3.794833  0.461951  0.535985  0.271260  0.262696  0.115692   

         Generosity  
group_1    0.317554  
group_2    0.197926  
group_3    0.220924  
group_4    0.246339  
group_5    0.232115

In [28]:
legend = ['group_1','group_2','group_3','group_4','group_5']
px.bar(summary_2015['Happiness Score'],y='Happiness Score',title='Mean Happiness Score:')

In [30]:
names = ['Economy','Family','Health','Freedom','Trust','Generosity']
fig = go.Figure()

for i in legend:
    fig.add_trace(go.Scatter(x=names,y=[summary_2016[j][i] for j in names],name=i))

fig.show()